runtime/opt/taupage/init.d/10-prepare-disks.py

```diff
@@ -105,16 +115,33 @@ def format_partition(partition, filesystem="ext4", initialize=False, is_already_
 
 def check_partition(partition, filesystem):
     if filesystem.startswith('ext'):
-        call = ['e2fsck', '-f', '-p']
+        call = ['e2fsck', '-f', '-p', partition]
+        wait_for_device(partition)
+        try:
+            call_command(call)
+        except CmdException as e:
+            # see e2fsck(8) man page for description of exit codes
+            if e.returncode <= 1:
+                # all OK
+                return
+            elif (e.returncode & 2) != 0:
+                logging.exception("File system errors corrected on %s, system should be rebooted.", partition)
+            elif (e.returncode & 4) != 0:
+                logging.exception("File system errors left uncorrected on %s.", partition)
+            elif e.returncode == 8:
```

Comment:
We already had an issue with exitcode 8 from e2fsck, which erased all our pipelines. 
I will change this to logging.exception, before merge.